### ISSUE TITLE
Add tests for normalizing `links` in Json and Rest serializer

### ISSUE THREAD

**Issue by igorT** [emberjs/data#3188]
We don't test what happens if you set a custom key in attrs for a relationship and receive that relationship with a link

**Comment by tpitale** [comment]
Do you have any pointers to where this happens in the code, so I can write the test for that?

**Comment by igorT** [comment]
Basically https://github.com/emberjs/data/blob/master/packages/ember-data/tests/integration/serializers/json-serializer-test.js#L222
but instead of  `my_comments: [1, 2]` pass in `links: { my_comments: 'url' }`
and assert that link got normalized correctly. This is a bug right now, so i would expect the test to fail fwiw

**Comment by tpitale** [comment]
Great, I should be able to do that.

**Comment by tim-evans** [comment]
@igorT I believe I wrote this. See https://github.com/emberjs/data/pull/1371 for docs on the current behavior

**Comment by igorT** [comment]
@tim-evans your pr doesn't normalize the link keys though or have tests for that. For example 
```
{ id:1 , links: { 'my_relationship': url }}
```
doesn't turn into 
```
{ id:1 , links: { 'myRelationship': url }}
```

**Comment by tim-evans** [comment]
Understood. I'm just making sure that people know where the things are related to this :)

**Comment by igorT** [comment]
@tim-evans ah, sorry i misparsed your answer, thought when you said `i wrote this` you were referring to this issue

**Comment by tim-evans** [comment]
It's all good :P

**Comment by pangratz** [comment]
Should be fixed by #4268.
